### PR TITLE
(hack) Convert guides to YAML and pretty print

### PR DESCRIPTION
**Pull request by lucywyman**
When we first implemented the `guide` subcommand, we chose to make the
file format of guides flat text and documented how users could add their
own guides to Bolt to be loaded by the `bolt guide` subcommand. While
useful, we haven't seen anyone use this functionality, and further it
does not seem like the file format has much impact on whether users add
their own guides or not. This converts the file format from flat text to
YAML, which supports both being read as data and formatting multiline
strings. This lets us format the `bolt guide <guide>` output to be more
readable and inline with other bolt output, make the guide text easily
editable (as opposed to JSON), and still lets users add their own guides
if they choose.

!feature

* **Beautify `bolt guide` output** (hack)
  This converts our existing guides from flat text to YAML so that they
  can be pretty printed.

### CODE REVIEW

**Comment by beechtom** [comment]
Since these now need to be parsed before printing, should the guides just be data in Bolt instead of separate files?

**Comment by lucywyman** [comment]
I don't think so - this still allows people to add their own guides if they'd like, plus having so much text as code is kind of awful, plus having these be JSON makes them more difficult to format so keeping them as YAML is preferable. I don't see any advantage to changing where they're loaded from.

File: guides/guide.txt [hunk 1]
```diff
@@ -1,17 +0,0 @@
-TOPIC
```
Comment (beechtom) [inline guides/guide.txt:1]: RIP 💀

File: guides/debugging.yaml [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-    debugging
-
-DESCRIPTION
+---
+topic: debugging
+guide: |
     When Bolt isn't behaving as expected, there are a few helpful commands and
     logs that can help identify common issues. The first place to look is in
-    `<PROJECT>/bolt-debug.log`, which contains debug-level logs from the last Bolt
+    `<PROJECT|/bolt-debug.log`, which contains debug-level logs from the last Bolt
```
Comment (beechtom) [inline guides/debugging.yaml:6]: Why the change to a pipe?

File: guides/logging.yaml [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
+guide: |
     Bolt prints messages both to the console and to log files. Messages can
     either come from Bolt's 'outputter', which logs user-facing messages like
     progress and results, or from the 'logger', which logs warnings, errors, and
     log-structured output to log files. Both of these message streams are
     configurable.
 
     By default, Bolt logs to the console at 'warn' level and writes a log file to
-    '<project>/bolt-debug.log' at 'debug' level. Unless you are running a plan,
+    '<project|/bolt-debug.log' at 'debug' level. Unless you are running a plan,
```
Comment (beechtom) [inline guides/logging.yaml:10]: Same here; why the change to a pipe?

File: guides/module.yaml [hunk 4]
```diff
@@ -14,6 +13,5 @@ DESCRIPTION
     
     To learn more about managing modules in a project, see the documentation.
     To learn how modules are loaded by Bolt, see the 'modulepath' guide.
-
-DOCUMENTATION
-    https://pup.pt/bolt-modules
+documentation:
```
Comment (beechtom) [inline guides/module.yaml:16]:
```suggestion

documentation:
```

Just a small nit for consistency, since the rest of the guides have an extra newline before documentation. 😬

File: guides/targets.yaml [hunk 5]
```diff
@@ -16,7 +15,7 @@ DESCRIPTION
          --rerun   FILTER       -Rerun        FILTER
 
     The 'targets' option accepts a comma-separated list of target URIs or group
-    names, or can read a target list from an input file '@<file>' or stdin '-'.
+    names, or can read a target list from an input file '@<file|' or stdin '-'.
```
Comment (beechtom) [inline guides/targets.yaml:18]: Why a pipe?

File: guides/transports.yaml [hunk 6]
```diff
@@ -1,13 +1,12 @@
...
+---
+topic: transports
+guide: |
     Bolt uses transports (also known as protocols) to establish a connection
     with a target in order to run actions on the target. The default transport is
     SSH, and you can see available transports along with their configuration
     options and defaults at http://pup.pt/bolt-reference.
 
-    You can specify a transport for a target by prepending '<transport>://' to
+    You can specify a transport for a target by prepending '<transport|://' to
```
Comment (beechtom) [inline guides/transports.yaml:9]: Pipe?

File: rakelib/pwsh.rake [hunk 7]
```diff
@@ -6,16 +6,33 @@ require 'fileutils'
...
+        # Make sure both topic and guide keys are defined
+        unless (%w[topic guide] - info.keys).empty?
+          raise "Guide file #{file} must have a 'topic' key and 'guide' key, but has #{info.keys} keys."
+        end
+
+        txt = +"#{info['topic']}\n"
+        txt << info['guide'].gsub(/^/, '  ')
+
+        if info['documentation']
+          txt << "\nDocumentation"
```
Comment (beechtom) [inline rakelib/pwsh.rake:28]:
```suggestion
          txt << "\nDocumentation\n"
```

File: guides/debugging.yaml (hunk 2 above)
Comment (lucywyman) [reply guides/debugging.yaml:6]: Oh, ha, definitely a typo.

File: guides/logging.yaml (hunk 3 above)
Comment (lucywyman) [reply guides/logging.yaml:10]: I think I used a sed command to find + replace that must have modified these....

File: guides/guide.txt (hunk 1 above)
Comment (lucywyman) [reply guides/guide.txt:1]: Gotta think of a better one :P

File: guides/logging.yaml [hunk 8]
```diff
@@ -1,7 +1,5 @@
-TOPIC
-    logging
-
-DESCRIPTION
+topic: logging
```
Comment (beechtom) [inline guides/logging.yaml:1]:
```suggestion
---
topic: logging
```

**Review by beechtom — APPROVED** [review]
